## Edit the values of the variables to fit your setup

# trimming scripts and primer files are from https://github.com/ItokawaK/Alt_nCov2019_primers
# edit these paths to point to the appropriate files
trimming_script="Alt_nCov2019_primers/tools/trim_primers/trim_primer_parts.py"
primer_file="Alt_nCov2019_primers/Primers/ver_N1/primer.bed"

# TODO: update annotation file to the one used in the publication
# saf file is generated by primer_count.R
primers_left_saf="primers_left.saf"

# specify your output folder here
output_dir="results"

# bwa index needs to be build and specified
index_bwa="indices/bwa/MN908947.3/MN908947.3.fa"
threads=8

ADAPTER_LEFT="CACGACGCTCTTCCGATCT"
ADAPTER_RIGHT="GACGTGTGCTCTTCCGATCT"

# analysis enspired by the repo from this paper https://github.com/ItokawaK/Alt_nCov2019_primers
analysis_paper (){
    fq1=$1
    fq2=$2
    out=$output_dir
    # TODO: try out if this works
    fq1_trimmed=${fq1%.fastq.gz}_trimmed.fastq.gz
    fq2_trimmed=${fq2%.fastq.gz}_trimmed.fastq.gz
    cutadapt -a $ADAPTER_FWD -A $ADAPTER_REV -o $fq1_trimmed -p $fq2_trimmed $fq1 $fq2

    # first pass aligment required for primer trimming as in the paper
    bwa mem -t $threads $index_bwa $fq1 $fq2 | tee $out/${fq1%_R1.fastq.gz}.sam | $trimming_script --gzip $primer_file ${fq1%_R1.fastq.gz}_trimmed.fastq ${fq2%.fastq.gz}_trimmed.fastq
    # second pass alignment with the primer trimmed reads
    bwa mem -t $threads $index_bwa ${fq1%_R1.fastq.gz}_trimmed.fastq.gz ${fq2%.fastq.gz}_trimmed.fastq.gz > $out/${fq1%_R1.fastq.gz}_trimmed.sam
    # sort and convert to bam
    samtools sort $out/${fq1%_R1.fastq.gz}.sam  > $out/${fq1%_R1.fastq.gz}.bam
    samtools index $out/${fq1%_R1.fastq.gz}.bam   
    rm -v $out/${fq1%_R1.fastq.gz}.sam

    samtools sort $out/${fq1%_R1.fastq.gz}_trimmed.sam  > $out/${fq1%_R1.fastq.gz}_trimmed.bam
    samtools index $out/${fq1%_R1.fastq.gz}_trimmed.bam
    rm -v $out/${fq1%_R1.fastq.gz}.sam

    # only count the non primer trimmed filed as the annotation file used 
    # contains the primer binding sites
    featureCounts -a $primers_left_saf -F SAF -o $out/${fq1%.fastq.gz}.counts -p -M --fraction -s 1 $out/${fq1%_R1.fastq.gz}.bam
}

